cli/src/torii/routing.rs

```diff
@@ -81,122 +79,31 @@ async fn handle_instructions(
-
-                let curr_cursor = Some(0);
-                let live_query = live_query.batched(fetch_size);
-                construct_query_response(
-                    request_id,
-                    &query_store,
-                    query_id,
-                    curr_cursor,
-                    live_query,
-                )
-            }
-        }
-    })
-}
-
-fn construct_query_response(
-    request_id: (&ValidQueryRequest, &Sorting, &Pagination),
-    query_store: &LiveQueryStore,
-    query_id: String,
-    curr_cursor: Option<u64>,
-    mut live_query: Batched<Vec<Value>>,
-) -> Result<Scale<BatchedResponse<Value>>> {
-    let (batch, next_cursor) = live_query.next_batch(curr_cursor)?;
-
-    if !live_query.is_depleted() {
-        query_store.insert(query_id.clone(), request_id, live_query);
-    }
-
-    let query_response = BatchedResponseV1 {
-        batch: Value::Vec(batch),
-        cursor: ForwardCursor {
-            query_id: Some(query_id),
-            cursor: next_cursor,
-        },
-    };
-
-    Ok(Scale(query_response.into()))
-}
-
-fn apply_sorting_and_pagination(
-    iter: impl Iterator<Item = Value>,
-    sorting: &Sorting,
-    pagination: Pagination,
-) -> Vec<Value> {
-    if let Some(key) = &sorting.sort_by_metadata_key {
-        let mut pairs: Vec<(Option<Value>, Value)> = iter
-            .map(|value| {
-                let key = match &value {
-                    Value::Identifiable(IdentifiableBox::Asset(asset)) => match asset.value() {
-                        AssetValue::Store(store) => store.get(key).cloned(),
-                        _ => None,
-                    },
-                    Value::Identifiable(v) => TryInto::<&dyn HasMetadata>::try_into(v)
-                        .ok()
-                        .and_then(|has_metadata| has_metadata.metadata().get(key))
-                        .cloned(),
-                    _ => None,
-                };
-                (key, value)
-            })
-            .collect();
-        pairs.sort_by(
-            |(left_key, _), (right_key, _)| match (left_key, right_key) {
-                (Some(l), Some(r)) => l.cmp(r),
-                (Some(_), None) => Ordering::Less,
-                (None, Some(_)) => Ordering::Greater,
-                (None, None) => Ordering::Equal,
-            },
-        );
-        pairs
-            .into_iter()
-            .map(|(_, val)| val)
-            .paginate(pagination)
-            .collect()
-    } else {
-        iter.paginate(pagination).collect()
-    }
+    sumeragi
+        .apply_wsv(|wsv| {
+            live_query_store.handle_query(
+                WithQueryParameters {
+                    query,
+                    fetch_size,
+                    sorting,
+                    pagination,
+                    cursor,
+                },
+                wsv,
+            )
+        })
+        .map(Scale)
+        .map_err(Into::into)
```

Comment:
Please rename `handle_query` to `handle_query_result`

I was just talking to @Erigara the other day how this function only uses `wsv` to execute the query. Could we make it like this:
```suggestion
    sumeragi
        .apply_wsv(|wsv| valid_request.execute(wsv).map_err(ValidationFail::from))
        .map(|query_result| {
            live_query_store.handle_query_result(
                WithQueryParameters {
                    query,
                    fetch_size,
                    sorting,
                    pagination,
                    cursor,
                },
                wsv,
            )
        })
        .map(Scale)
        .map_err(Into::into)
```

Reply:
I propose to construct API in such a way that `apply_wsv` takes as little time as possible to execute. In this case I would just like to extract `valid_request.execute(wsv)` out of `handle_query`

Reply:
Ok, I see, makes sense. I'll do that

Reply:
I did it the way you proposed, but `LazyQuery` returned by a query execution depends on the `wsv` lifetime, so it's not possible.

However I removed validation/execution from `LiveQueryStoreHandle`.

Reply:
> I did it the way you proposed, but LazyQuery returned by a query execution depends on the wsv lifetime, so it's not possible.

yeah, I know. As far as I am aware this lifetime bound cannot be expressed in Rust. But still it's better you have pulled it out